File: app/src/main/java/com/esbati/keivan/persiancalendar/POJOs/CalendarRemark.java
package com.esbati.keivan.persiancalendar.POJOs;

import org.json.JSONException;
import org.json.JSONObject;

import ir.smartlab.persindatepicker.util.PersianCalendar;

/**
 * Created by asus on 11/21/2016.
 */

public class CalendarRemark {

    public String mTitle;
    public int mYear;
    public int mMonth;
    public int mDay;
    public boolean isHoliday;
    public PersianCalendar mPersianDate;

    public CalendarRemark(){

    }

    public CalendarRemark fromJSON(JSONObject eventJSON) throws JSONException{
        mTitle = eventJSON.getString("title");
        mYear = eventJSON.optInt("year", -1);
        mMonth = eventJSON.getInt("month");
        mDay = eventJSON.getInt("day");
        isHoliday = eventJSON.getBoolean("holiday");
        mPersianDate = new PersianCalendar().setPersianDate(mYear, mMonth, mDay);
        return this;
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/pojos/CalendarRemark.kt
package com.esbati.keivan.persiancalendar.pojos

import org.json.JSONException
import org.json.JSONObject

/**
 * Created by asus on 11/21/2016.
 */

data class CalendarRemark (
    val mTitle: String,
    val mYear: Int,
    val mMonth: Int,
    val mDay: Int,
    val isHoliday: Boolean) {

    fun inTheSameDate(year: Int, month: Int, day: Int): Boolean {
        return (mYear == -1 || mYear == year) && mMonth == month && mDay == day
    }

    companion object {
        @Throws(JSONException::class)
        fun fromJSON(eventJSON: JSONObject): CalendarRemark {
            val title = eventJSON.getString("title")
            val year = eventJSON.optInt("year", -1)
            val month = eventJSON.getInt("month")
            val day = eventJSON.getInt("day")
            val isHoliday = eventJSON.getBoolean("holiday")
            return CalendarRemark(title, year, month, day, isHoliday)
        }
    }
}
